backend/test/integrated/mitigation.js

```diff
@@ -0,0 +1,335 @@
+			'General steelwork',
+			'Stability and erection of structural steelwork',
+			'Masonry',
+			'Timber',
+			'Refurbishment of existing buildings',
+			'External Cladding',
+			'Roof coverings and finishes',
+			'Atria',
+			'Windows/glazing including windows cleaning',
+			'Surface coating and finishes',
+			'Cleaning of buildings',
+			'Mechanical services',
+			'Electrical services',
+			'Public health services',
+			'Lifts, escalators and auto walks',
+			'General civil engineering, including small works',
+			'Roads, working adjacent to, maintenance of',
+			'Railways, working adjacent to, maintenance of',
+			'Bridge construction',
+			'Bridge maintenance',
+			'Working over/near water',
+			'Pipes and cables',
+			'Work in coastal and maritime waters'
+		],
+		associated_activity: [
+			'Install construction',
+			'Preliminary investigation, tests & protypes',
+			'Material sourcing',
+			'Component manufacture',
+			'Storage, transport, logistics',
+			'Commission; site tests',
+			'Use',
+			'Operation',
+			'Maintenance',
+			'High impact events',
+			'Modification',
+			'Aging',
+			'Life extension',
+			'Demolition, removal',
+			'Post processing',
+			'Material disposal or re-use',
+			'Not Applicable',
+			'0'
+		]
+	};
+
+	before(function(done) {
+
+		server = app.listen(8080, function () {
+			agent = request.agent(server);
+			console.log("API test server is listening on port 8080!");
+			done();
+		});
+	});
+
+	after(function(done) {
+		server.close(function() {
+			console.log("API test server is closed");
+			done();
+		});
+	});
+
+	describe("Get mitigation criteria", function(done) {
+		before(function(done) {
+			this.timeout(timeout);
+			agent.post("/login")
+				.send({username, password})
+				.expect(200, done);
+
+		});
+
+		it("should succeed", function(done) {
+			agent.get(`/${username}/mitigations/criteria`)
+				.expect(200, function(err, res) {
+					expect(res.body).to.deep.equal(goldenCriteria);
+					done(err);
+				});
+		});
+
+		after(function(done) {
+			this.timeout(timeout);
+			agent.post("/logout")
+				.expect(200, done);
+		});
+	});
+
+	describe("Find mitigation suggestions", function(done) {
+		const totalSuggestions = 201;
+		const expectedLengths = {
+			"category": 162,
+			"location_desc": 46,
+			"element": 84,
+			"risk_factor": 115,
+			"scope": 163,
+			"associated_activity": 163
+		};
+		const invalidCriteria = {
+			"mitigation_desc": "Avoid trap hazards near openings.",
+			"mitigation_detail": "Detailed information",
+			"mitigation_stage": goldenCriteria["mitigation_stage"][0],
+			"mitigation_type": goldenCriteria["mitigation_type"][0],
+			"abcdefghijklmnop": "qrstuvwxyz",
+			"invalidFieldName": "randomData"
+		};
+
+		before(function(done) {
+			this.timeout(timeout);
+			agent.post("/login")
+				.send({username, password})
+				.expect(200, done);
+
+		});
+
+		it("without criteria should succeed", function(done) {
+			agent.post(`/${username}/mitigations`)
```

Comment:
Again:

- if username doesn't exist
- If user is not part of teamspace
- if user is not admin